pkg/cmd/roachprod/flags.go

```diff
@@ -273,6 +273,8 @@ func initFlags() {
 	deployCmd.Flags().DurationVar(&pause, "pause", pause, "duration to pause between node restarts")
 
 	syncCmd.Flags().BoolVar(&listOpts.IncludeVolumes, "include-volumes", false, "Include volumes when syncing")
+	syncCmd.Flags().StringArrayVarP(&listOpts.IncludeProviders, "clouds", "c",
+		make([]string, 0), "Include the specific cloud providers when syncing")
```

Comment:
One concern I realised with this change is if someone uses this flag on the default ephemeral project it may cause issues for other people, until someone does a proper sync again. Not sure if we should include it here as a warning?  The help message could maybe be a bit more clear in that it says only the specified providers will be synced, and the default is all available providers.